src/com/lftechnology/java/training/krishna/employeemanagement/domain/Employee.java

```diff
@@ -0,0 +1,128 @@
+
+package com.lftechnology.java.training.krishna.employeemanagement.domain;
+
+/**
+ * This program store employee information
+ * 
+ * @author Krishna Timilsina <krishnatimilsina@lftechnology.com>
+ * @version 1.0
+ */
+public class Employee extends User {
+
+	private String fullname;
+	private String department;
+	private String address;
+	private String role;
+
+	//constructor
+	public Employee() {
+
+	}
+
+
+	public String getFullname() {
+
+		return fullname;
+	}
+
+	public void setFullname(String fullname) {
+
+		this.fullname = fullname;
+	}
+
+	public String getDepartment() {
+
+		return department;
+	}
+
+	public void setDepartment(String department) {
+
+		this.department = department;
+	}
+
+	public String getAddress() {
+
+		return address;
+	}
+
+	public void setAddress(String address) {
+
+		this.address = address;
+	}
+
+	public String getRole() {
+
+		return role;
+	}
+
+	public void setRole(String role) {
+
+		this.role = role;
+	}
+
+
+
+	@Override
+	public String toString() {
+
+		return "Employee [fullname=" + fullname + ", department=" + department +
+			", address=" + address + ", role=" + role + ", getId()=" + getId() +
+			", getUsername()=" + getUsername() + ", getPassword()=" +
+			getPassword() + ", getIsTerminated()=" + getIsTerminated() +
+			", getCreatedAt()=" + getCreatedAt() + ", getModifiedAt()=" +
+			getModifiedAt() + ", toString()=" + super.toString() +
+			", getClass()=" + getClass() + "]";
+	}
+
+	@Override
+	public int hashCode() {
+
+		final int prime = 31;
+		int result = super.hashCode();
+		result = prime * result + ((address == null) ? 0 : address.hashCode());
+		result =
+			prime * result + ((department == null) ? 0 : department.hashCode());
+		result =
+			prime * result + ((fullname == null) ? 0 : fullname.hashCode());
+		result = prime * result + ((role == null) ? 0 : role.hashCode());
+		return result;
+	}
+
+	@Override
+	public boolean equals(Object obj) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "equals" is 27 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)